### PR TITLE
Limit total size of http-client.log

### PR DESCRIPTION
**Pull request by gjhkael**
Fixes [894](https://github.com/airlift/airlift/issues/894)
Fixes https://github.com/trinodb/trino/issues/7018

### CODE REVIEW

File: http-client/src/main/java/io/airlift/http/client/jetty/DefaultHttpClientLogger.java [hunk 1]
```diff
@@ -67,6 +68,10 @@
             rollingPolicy.setFileNamePattern(rollingPolicy.getFileNamePattern() + ".gz");
         }
 
+        // Limit total log files occupancy on disk. Ideally we would keep exactly
```
Comment (findepi) [inline http-client/src/main/java/io/airlift/http/client/jetty/DefaultHttpClientLogger.java:71]: Also update `rollingPolicy.setMaxHistory(maxHistory);` line above, with `// limits number of logging periods (i.e. days) kept` comment

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Can you reference b6ce4e490a76c6d230d414cf410baabd18c541e7 in the commit message, for future reference?

File: http-client/src/main/java/io/airlift/http/client/jetty/DefaultHttpClientLogger.java (hunk 1 above)
Comment (gjhkael) [reply http-client/src/main/java/io/airlift/http/client/jetty/DefaultHttpClientLogger.java:71]: got it

**Comment by gjhkael** [comment]
Do I need to merge the two commits?

**Comment by gjhkael** [comment]
The checks failed seems not my problem.

**Comment by findepi** [comment]
> Do I need to merge the two commits?

yes please. 
this will also retrigger the build. unfortnately there are some flaky tests, sorry for that

**Comment by gjhkael** [comment]
After rebase the two commit finished in local, when I push to remote repo, the github notify me to pull the remote branch first. Then the number of commits become four. So I delete the remote branch and push again. May be I need push -f ?

**Pull request closed without merging**